src/Angor/Avalonia/Angor.Wallet/Infrastructure/Impl/WalletAppService.cs

```diff
@@ -0,0 +1,266 @@
+            }
+
+            var (seed, passphrase) = sensitiveDataResult.Value;
+            
+            var walletWords = new WalletWords { Words = seed, Passphrase = passphrase };
+            var accountInfo = _walletOperations.BuildAccountInfoForWalletWords(walletWords);
+        
+            // Update AccountInfo to ensure it has change addresses
+            await _walletOperations.UpdateAccountInfoWithNewAddressesAsync(accountInfo);
+        
+            var changeAddress = accountInfo.GetNextChangeReceiveAddress();
+            if (string.IsNullOrEmpty(changeAddress))
+                return Result.Failure<Fee>("No change address available");
+
+            var sendInfo = new SendInfo
+            {
+                SendAmount = amount.Value,
+                SendToAddress = address.Value,
+                ChangeAddress = changeAddress,
+                SendUtxos = _walletOperations.FindOutputsForTransaction(amount.Value, accountInfo)
+                    .ToDictionary(data => data.UtxoData.outpoint.ToString(), data => data),
+            };
+
+            var estimatedFee = _walletOperations.CalculateTransactionFee(sendInfo, accountInfo, feeRate.SatsPerVByte);
+            return Result.Success(new Fee((long)(estimatedFee * 100_000_000))); // Convertir BTC a sats
+        }
+        catch (Exception ex)
+        {
+            return Result.Failure<Fee>($"Error estimating fee: {ex.Message}");
+        }
+    }
+    public async Task<Result<Address>> GetNextReceiveAddress(WalletId walletId)
+    {
+        if (walletId != SingleWalletId)
+            return Result.Failure<Address>("Invalid wallet ID");
+
+        try
+        {
+            var sensitiveDataResult = await sensitiveWalletDataProvider.RequestSensitiveData(walletId);
+            if (sensitiveDataResult.IsFailure)
+            {
+                return Result.Failure<Address>(sensitiveDataResult.Error);
+            }
+
+            var (seed, passphrase) = sensitiveDataResult.Value;
+            var walletWords = new WalletWords { Words = seed, Passphrase = passphrase };
+            var accountInfo = _walletOperations.BuildAccountInfoForWalletWords(walletWords);
+            await _walletOperations.UpdateAccountInfoWithNewAddressesAsync(accountInfo);
+
+            var address = accountInfo.GetNextReceiveAddress();
+            if (string.IsNullOrEmpty(address))
+                return Result.Failure<Address>("No address available");
+
+            return Result.Success(new Address(address));
+        }
+        catch (Exception ex)
+        {
+            return Result.Failure<Address>($"Error getting receive address: {ex.Message}");
+        }
+    }
+
+    public async Task<Result<TxId>> SendAmount(WalletId walletId, Amount amount, Address address, DomainFeeRate feeRate)
+    {
+        if (walletId != SingleWalletId)
+            return Result.Failure<TxId>("Invalid wallet ID");
+        
+        var finalFeeRate = feeRate.SatsPerVByte * 10000; 
```

Comment:
Why is this multiplied? if I understand correctly it should be * 1000 for sats per kilobyte

Reply:
I tried to copy the behavior of the UI, but I can completely wrong.

Given this:

![image](https://github.com/user-attachments/assets/640d2d87-e7c9-4859-9cd8-ff633d0b8ab2)

The sendInfo is this:

![image](https://github.com/user-attachments/assets/603b4ed4-ccbd-4b55-9998-e49ab0c18a30)

I'm not sure about the conversion, to be honest. What's the correct way to calculate?

Reply:
I think the sendinfo in this case has a high fee to get it in the next block, but if it is sats ber byte and we convert it to sats per kilobyte it should be * 1000

Reply:
Fixed!